### PR TITLE
2 cleanup commits

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -37,6 +38,7 @@ linter:
     - implementation_imports
     - invariant_booleans
     - iterable_contains_unrelated_type
+    - join_return_with_assignment
```
Comment (jakemac53) [inline analysis_options.yaml]: this style just looks weird to me but :man_shrugging: if it cuts down on code review churn I guess ill take it

File: json_serializable/test/custom_configuration_test.dart [hunk 2]
```diff
@@ -19,7 +19,7 @@ import 'shared_config.dart';
 
 LibraryReader _libraryReader;
 
-void main() async {
+Future main() async {
```
Comment (jakemac53) [inline json_serializable/test/custom_configuration_test.dart]: Ya, I would remove this lint lol these do not provide value

File: analysis_options.yaml [hunk 3]
```diff
@@ -23,6 +23,7 @@ linter:
     - avoid_types_as_parameter_names
     - avoid_unused_constructor_parameters
     - await_only_futures
+    - avoid_void_async
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply analysis_options.yaml]: ack

File: json_serializable/test/custom_configuration_test.dart (hunk 2 above)
Comment (kevmoo) [reply json_serializable/test/custom_configuration_test.dart]: ack

**Comment by kevmoo** [comment]
PTAL

**Pull request merged**